specs/Spec.Blake2.fst

```diff
@@ -419,7 +421,7 @@ val blake2_update_blocks:
   -> s:state a ->
   Tot (state a)
 
-let split a len =
+let split (a:alg) (len:nat) : nb_rem:(nat & nat){let (nb,rem) = nb_rem in nb * size_block a + rem == len} =
```

Comment:
should this move to a spec helper module or something? surely other algorithms must need this

Reply:
This particular split function is different from the one used in other algorithms, because Blake2 treats a *full* final block differently. Using a generic split here led to the previous bug, so trying to be very explicit here.

Reply:
sounds good